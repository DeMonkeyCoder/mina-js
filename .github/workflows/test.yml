name: Run tests
on:
    push:
        branches:
            - main
jobs:
    my-job:
        name: my-job
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v4
            - uses: oven-sh/setup-bun@v2
            - run: bun install
            - run: bun run build
            - run: bun run test
