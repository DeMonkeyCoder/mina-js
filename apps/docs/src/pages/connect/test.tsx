import { TestZkApp } from "../../components/test-zkapp";

const TestPage = () => {
	if (import.meta.env.SSR) return null;
	return <TestZkApp />;
};

export default TestPage;
