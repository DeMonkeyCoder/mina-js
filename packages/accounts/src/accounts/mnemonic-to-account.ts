import { HDKey } from "@scure/bip32";
import { mnemonicToSeedSync } from "@scure/bip39";

import type { HDAccount, HDOptions } from "../types";
import {
	type HDKeyToAccountOptions,
	hdKeyToAccount,
} from "./hd-key-to-account.js";

export type MnemonicToAccountOptions = HDKeyToAccountOptions;

/**
 * @description Creates an Account from a mnemonic phrase.
 *
 * @returns A HD Account.
 */
export function mnemonicToAccount(
	mnemonic: string,
	opts: HDOptions = {},
): HDAccount {
	const seed = mnemonicToSeedSync(mnemonic);
	return hdKeyToAccount(HDKey.fromMasterSeed(seed), opts);
}
