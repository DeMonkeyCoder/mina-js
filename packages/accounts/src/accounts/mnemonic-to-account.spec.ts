import { expect, it } from "bun:test";
import { mnemonic } from "../test/constants";
import { mnemonicToAccount } from "./mnemonic-to-account";

it("matches the snapshot", () => {
	const mnemonicAccount = mnemonicToAccount(mnemonic);
	expect(mnemonicAccount).toMatchSnapshot();
});
