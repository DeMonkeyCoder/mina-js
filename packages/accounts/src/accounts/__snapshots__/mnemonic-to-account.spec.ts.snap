// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
{
  "createNullifier": [Function: AsyncFunction],
  "getHdKey": [Function],
  "publicKey": "B62qkYa1o6Mj6uTTjDQCob7FYZspuhkm4RRQhgJg9j4koEBWiSrTQrS",
  "signFields": [Function: AsyncFunction],
  "signMessage": [Function: AsyncFunction],
  "signTransaction": [Function: AsyncFunction],
  "source": "hd",
  "type": "local",
}
`;
